.github/workflows/ci.yml

```
@@ -62,6 +62,19 @@ jobs:
       with:
         command: test
         args: --verbose --all-targets
+    
+    - name: Notify if Job Fails
+      uses: ravsamhq/notify-slack-action@v1
+      if: always() && github.ref == 'refs/heads/master'
```

Is the `always() &&` here actually needed?
```suggestion
      if: github.ref == 'refs/heads/master'
```

---

@adlerjohn - the always() is needed is there is only failures in the prior steps so the slack notification step is run in the job (if the past steps of a job successful - it does not trigger a slack notification) 

[
](https://docs.github.com/en/actions/learn-github-actions/expressions#:~:text=succeeded%0A%20%20%20%20if%3A%20%24%7B%7B%20success()%20%7D%7D-,always,the%20task%20from%20running.%20For%20example%2C%20if%20getting%20sources%20failed.,-Example)

I tested the recommendation you suggested but it did not trigger a notify after a prior step failed (on the build job and for my branch) as the always() was missing

The alternative is to try "failure()" is we do not want to use "always()" - which I can try

https://docs.github.com/en/actions/learn-github-actions/expressions#:~:text=if%3A%20%24%7B%7B%20cancelled()%20%7D%7D-,failure,dependent%20jobs%2C%20failure()%20returns%20true%20if%20any%20ancestor%20job%20fails.,-Example

---

I see. Okay.